BulkyWeb/Controllers/CategoryController.cs

```diff
@@ -16,5 +16,83 @@ public IActionResult Index()
             List<Category> objCategoryList = _db.Categories.ToList();
             return View(objCategoryList);
         }
+
+        public IActionResult Create()
+        {
+            return View();
+        }
+        [HttpPost]
+        public IActionResult Create(Category obj)
+        {
+            if (obj.Name == obj.DisplayOrder.ToString())
+            {
+                ModelState.AddModelError("name", "The DisplayOrder cannot exactly match the Name.");
+            }
+            if (ModelState.IsValid)
+            {
+                _db.Categories.Add(obj);
+                _db.SaveChanges();
+                TempData["success"] = "Category created successfully";
+                return RedirectToAction("Index");
+            }
+            return View();
+        }
+
+        public IActionResult Edit(int? id)
```

Comment:
The use of ? is to be able to pass in null values. So putting int ? id allows for null values to be passed to the controller and then you have to check if its null when you can just change it to int id and that will not allow null values straight from the url

Reply:
I see a lot in the code so you will need to refactor tyhe code and include more comments. I will say pick a commenting style to be using moving forward

Reply:
Noted